modules/form-field/react/lib/Hint.tsx

```diff
@@ -19,9 +19,20 @@ export interface HintProps extends Themeable, React.HTMLAttributes<HTMLParagraph
   alertLabel?: string;
 }
 
-const Label = styled('span')(type.body2, type.variant.label);
+const Label = styled('span')<Pick<HintProps, 'error'>>(
+  type.body2,
+  type.variant.label,
+  ({error}) => error === ErrorType.Error && {color: statusColors.error}
```

Comment:
Looking at the semantic groupings we have for colors, `statusColors.error` seems to make the most sense. Let me know if I should be referencing a different semantic color (or even `cinnamon500` explicitly).